src/Nethermind/Nethermind.TxPool/Collections/DistinctValueSortedPool.cs

```diff
@@ -16,47 +16,53 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Nethermind.TxPool.Collections
 {
-    public class DistinctValueSortedPool<TKey, TValue> : SortedPool<TKey, TValue>
+    public class DistinctValueSortedPool<TKey, TValue, TGroup> : SortedPool<TKey, TValue, TGroup>
     {
-        private readonly IDictionary<TValue, LinkedListNode<KeyValuePair<TKey, TValue>>> _distinctDictionary;
+        private readonly IComparer<TValue> _comparer;
+        private readonly IDictionary<TValue, KeyValuePair<TKey, TValue>> _distinctDictionary;
 
-        public DistinctValueSortedPool(int capacity, Comparison<TValue> comparison, IEqualityComparer<TValue> distinctComparer) : base(capacity, comparison)
+        public DistinctValueSortedPool(
+            int capacity,
+            IComparer<TValue> comparerWithIdentity,
+            Func<TValue, TGroup> groupMapping,
+            IEqualityComparer<TValue> distinctComparer,
+            IComparer<TValue> comparer) 
```

Comment:
there are three different comparers coming to this class?

Reply:
I found it difficult to follow the constructor arguments meaning.

Reply:
Adding comments